### PR TITLE
Dev - Bugfixes for Field of Ruin & Cycle of Rebirth

### PR DESCRIPTION
**Pull request by marcsoiferman**
Handles Field of Ruin not discarding opponent's cards.

Handles Cycle of Rebirth not shuffling opponent's deck.

### CODE REVIEW

File: server/game/cards/14.1-RaW/CycleOfRebirth.js [hunk 1]
```diff
@@ -13,16 +13,44 @@ class CycleOfRebirth extends DrawCard {
...
-                    shuffle: true
-                })),
+                AbilityDsl.actions.conditional({
+                    condition: context => context.target.controller === context.source.controller,
+                    trueGameAction: AbilityDsl.actions.moveCard(context => ({
+                        destination: Locations.DynastyDeck,
+                        target: [context.target, context.source],
+                        shuffle: true
+                    })),
+                    falseGameAction: AbilityDsl.actions.multiple([
```
Comment (Syrail) [inline server/game/cards/14.1-RaW/CycleOfRebirth.js]: Why have a conditional when the falseGameAction branch would cover both cases?

File: server/game/cards/14.1-RaW/CycleOfRebirth.js [hunk 2]
```diff
@@ -13,16 +13,44 @@ class CycleOfRebirth extends DrawCard {
...
+                        }))
+                    ])
+                }),
                 AbilityDsl.actions.refillFaceup(context => ({
                     target: [context.target.controller, context.source.controller],
                     location: [Locations.ProvinceOne, Locations.ProvinceTwo, Locations.ProvinceThree, Locations.ProvinceFour]
                 }))
-            ])
+            ]),
+            effect: 'shuffle {1}{3}{4} into {2}\'s dynasty deck{5}{6}{7}{8}{9}',
```
Comment (Syrail) [inline server/game/cards/14.1-RaW/CycleOfRebirth.js]: Putting the ternary in effect instead of effectArgs would make this easier to follow (fewer args).

Comment (marcsoiferman) [reply server/game/cards/14.1-RaW/CycleOfRebirth.js]: effect doesn't take the context as an arg, so I don't know how to put the conditional in the effect itself.

**Comment by marcsoiferman** [comment]
I think this closes #258

**Pull request merged**